filebeat/inputsource/tcp/scan.go

```diff
@@ -0,0 +1,31 @@
+package tcp
+
+import (
+	"bufio"
+	"bytes"
+)
+
+// ScanDelimiter return a function to split line using a custom delimiter, the delimiter
+// is stripped from the returned value.
+func scanDelimiter(delimiter []byte) bufio.SplitFunc {
```

Comment:
Is this similar to the strip_newline reader?

Reply:
Not really, I will rename this method to `factoryDelimiter()` which is what it does.

Reply:
It returns a custom `split func` that support a custom delimiter which can be a multibytes delimiter.